src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -379,11 +379,12 @@ public void Initialize(ITelemetry telemetry)
 
             if (string.IsNullOrEmpty(instrumentationKey))
             {
-                instrumentationKey = PlatformSingleton.Current.GetEnvironmentVariable(InstrumentationKeyWebSitesEnvironmentVariable);
-                
-                if (string.IsNullOrEmpty(instrumentationKey))
+                // If no configuration override was passed to the constructor then the default configuration instance will have been used.
+                // If no iKey was present in the config then the environment variable will have been loaded into the default instance.
+                instrumentationKey = this.configuration.InstrumentationKey;
+                if (string.IsNullOrEmpty(instrumentationKey) && this.configuration != TelemetryConfiguration.Active)
```

Comment:
So the decision is to use the ikey from the `ApplicaitonInsights.config` to take priority? `TelemtryConfigurationFactory` seems to kept this logic. Can you please add a test for this case or just point to it